server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -371,9 +372,12 @@ public class Node implements Closeable {
         }
     }, Setting.Property.NodeScope);
 
-    public static final Setting<ByteSizeValue> NODE_SEARCH_CACHE_SIZE_SETTING = Setting.byteSizeSetting(
+    private static final String ZERO = "0";
+
+    public static final Setting<String> NODE_SEARCH_CACHE_SIZE_SETTING = new Setting<>(
         "node.search.cache.size",
-        ByteSizeValue.ZERO,
+        s -> DiscoveryNode.isDedicatedSearchNode(s) ? "80%" : ZERO,
```

Comment:
Actually, I was planning to validate the node role here. I would throw an exception if the user doesn't set the cache size explicitly and this node is not a dedicated search node. However, I found that this default value function is always called, regardless of whether we set the setting or not. Maybe we can change this behavior later (call the value function only when necessary) and add validation here.

https://github.com/opensearch-project/OpenSearch/blob/10ae4cb97f0616b1728d1ba07b2d9ae8601071e4/server/src/main/java/org/opensearch/common/settings/Setting.java#L569